src/dotnet/ToolPackage/ToolPackageObtainer.cs

```diff
@@ -95,16 +96,47 @@ IProjectRestorer projectRestorer
                 packageVersion = concreteVersion;
             }
 
-            ToolConfiguration toolConfiguration = GetConfiguration(packageId: packageId, packageVersion: packageVersion, individualToolVersion: toolDirectory);
+            LockFile lockFile = new LockFileFormat()
+                .ReadWithLock(toolDirectory.WithFile("project.assets.json").Value)
+                .Result;
 
-            return new ToolConfigurationAndExecutableDirectory(
+            LockFileItem dotnetToolSettings = FindAssetInLockFile(lockFile, "DotnetToolSettings.xml", packageId);
+
+            if (dotnetToolSettings == null)
+            {
+                throw new PackageObtainException("Cannot find DotnetToolSettings");
+            }
+
+            FilePath toolConfigurationPath =
+                toolDirectory
+                    .WithSubDirectories(packageId, packageVersion)
+                    .WithFile(dotnetToolSettings.Path);
+
+            ToolConfiguration toolConfiguration =
+                ToolConfigurationDeserializer.Deserialize(toolConfigurationPath.Value);
+
+            var entryPointFromLockFile = FindAssetInLockFile(lockFile, toolConfiguration.ToolAssemblyEntryPoint, packageId);
```

Comment:
Is there a reason we need to specifically look for the entry point in the lockfile?  Could we just combine the path of the directory the settings file is in with the relative path to the entry point in that settings file?

Reply:
what in nuget is "tools/tfm/rid/my/tools.dll" what we need from nuget is "tools/tfm/rid" in your case, however, we need to match tfm and rid pattern to get that. I think it is easier to match entrypoint directly